enclave_build/src/docker.rs

```diff
@@ -174,6 +174,13 @@ impl DockerUtil {
     /// Pull the image, with the tag provided in constructor, from the Docker registry
     pub fn pull_image(&self) -> Result<(), DockerError> {
         let act = async {
+            // Check if the Docker image is locally available.
+            // If available, early exit.
+            if let Ok(_image) = self.docker.images().get(&self.docker_image).inspect().await {
```

Comment:
Nit: will the `.await` at the end force the evaluation of the yielded `Future` object, producing a `Result`? If this is indeed the case and the outcome is a result, could we call `.is_ok()` on it?

Something like `if self.docker.images().get(&self.docker_image).inspect().await.is_ok() { ... }`, since the `_image` wrapped inside the `Ok` is not used?

Reply:
Yup, that will work, thanks for the note. I updated the logic.